### PR TITLE
feat: show successful resource delete message

### PR DESCRIPTION
**Pull request by dbolson**
We needed to refactor some things to support showing a "Successfully deleted {resource}" message because we don't get a JSON response from API delete requests.

**Requirements**

- [ ] I have added test coverage for new or changed functionality
- [ ] I have followed the repository's [pull request submission guidelines](../blob/main/CONTRIBUTING.md#submitting-pull-requests)
- [ ] I have validated my changes against all supported platform versions

**Related issues**

Provide links to any issues in this repository or elsewhere relating to this pull request.

**Describe the solution you've provided**

Provide a clear and concise description of what you expect to happen.

**Describe alternatives you've considered**

Provide a clear and concise description of any alternative solutions or features you've considered.

**Additional context**

Add any other context about the pull request here.

### CODE REVIEW

File: cmd/flags/create.go [hunk 1]
```diff
@@ -81,10 +81,9 @@ func runCreate(client flags.Client) func(*cobra.Command, []string) error {
 			return errors.NewError(output)
 		}
 
-		output, err := output.CmdOutputCreate(
+		output, err := output.CmdOutputCreateResource(
```
Comment (dbolson) [inline cmd/flags/create.go:84]: The API has one less param now.

File: internal/output/resource_output.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+func (r resourceOutput) plaintext() string {
+	return r.outputFn(r.resource)
+}
+
+type resourceOutputFn func(input []byte, outputFn PlaintextOutputFn, r resource) resourceOutput
+
+// CmdOutputCreateResource returns a response from a resource create action formatted based on the
+// output flag.
+func CmdOutputCreateResource(outputKind string, input []byte) (string, error) {
```
Comment (dbolson) [inline internal/output/resource_output.go:31]: The create and update versions are very similar but delete is different.

**Review by sunnyguduru — APPROVED** [review]

File: cmd/environments/get.go [hunk 3]
```diff
@@ -76,11 +76,7 @@ func runGet(
 			return errors.NewError(output)
 		}
 
-		output, err := output.CmdOutputSingular(
-			viper.GetString(cliflags.OutputFlag),
-			response,
-			output.SingularPlaintextOutputFn,
-		)
+		output, err := output.CmdOutput("get", viper.GetString(cliflags.OutputFlag), response)
```
Comment (dbolson) [inline cmd/environments/get.go:79]: We need to know if the plaintext output should include a success message, so we pass in this value.

File: internal/members/members.go [hunk 4]
```diff
@@ -42,10 +42,10 @@ func (c MembersClient) Create(ctx context.Context, accessToken string, baseURI s
 	if err != nil {
 		return nil, errors.NewLDAPIError(err)
 	}
-	memberJson, err := json.Marshal(members.Items)
+	membersJson, err := json.Marshal(members)
```
Comment (dbolson) [inline internal/members/members.go:45]: Updated this for consistency with other responses for multiple resources.

File: internal/output/output.go [hunk 5]
```diff
@@ -73,33 +72,7 @@ func CmdOutputSingular(outputKind string, input []byte, fn PlaintextOutputFn) (s
...
-	return outputFromKind(outputKind, "", SingularOutputter{
-		outputFn:     fn,
-		resource:     r,
-		resourceJSON: input,
-	})
-}
-
-// CmdOutputCreate builds a command response based on the flag the user provided and the shape of
-// the input with a successfully created message. The expected shape is a single JSON object.
-func CmdOutputCreate(outputKind string, input []byte, fn PlaintextOutputFn) (string, error) {
```
Comment (dbolson) [inline internal/output/output.go:85]: Removed these to have a single `CmdOutput` function so it's simpler for dynamic resource commands.

File: internal/output/output.go [hunk 6]
```diff
@@ -73,67 +69,19 @@ func CmdOutputSingular(outputKind string, input []byte, fn PlaintextOutputFn) (s
...
+	return outputFromKind(outputKind, SingularOutputter{
 		outputFn:     fn,
 		resource:     r,
 		resourceJSON: input,
 	})
 }
 
-// CmdOutputMultiple builds a command response based on the flag the user provided and the shape of
-// the input. The expected shape is a list of JSON objects.
-func CmdOutputMultiple(outputKind string, input []byte, fn PlaintextOutputFn) (string, error) {
```
Comment (dbolson) [inline internal/output/output.go:111]: This is now unused.

File: internal/output/resource_output.go [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// CmdOutput returns a response from a resource create action formatted based on the
+// output flag along with an optional message based on the action.
+func CmdOutput(action string, outputKind string, input []byte) (string, error) {
```
Comment (dbolson) [inline internal/output/resource_output.go:11]: This is the "do all the things" function.

**Review by dbolson — COMMENTED** [review]
I can simplify the output function for error handling in another PR.

**Pull request merged**